packages/svelteui-core/src/components/Chip/Chip.styles.ts

```diff
@@ -0,0 +1,189 @@
+import { createStyles } from '$lib/styles';
+import type { DefaultProps, SvelteUIColor, SvelteUINumberSize, SvelteUISize } from '$lib/styles';
+import type { Component } from '$lib/internal';
+
+export interface ChipProps extends DefaultProps {
+	color?: SvelteUIColor;
+	id?: string;
+	checked?: boolean;
+	disabled?: boolean;
+	value?: string;
+	label?: Component | string;
+	radius?: SvelteUINumberSize | number;
+	size?: SvelteUISize;
+  type?: 'checkbox' | 'radio';
+  variant?: 'outline' | 'filled'
+	transitionDuration?: number;
+	wrapperProps?: { [key: string]: any };
+}
+
+export interface ChipStyleParams {
+	size: SvelteUISize;
+	color: SvelteUIColor;
+  variant: 'outline' | 'filled'
+	radius: SvelteUINumberSize | number;
+	transitionDuration: number;
+}
+
+export const sizes = {
+  xs: 24,
+  sm: 28,
+  md: 32,
+  lg: 36,
+  xl: 40,
+};
+
+const iconSizes = {
+  xs: 10,
+  sm: 12,
+  md: 14,
+  lg: 16,
+  xl: 18,
+};
+
+const padding = {
+  xs: 16,
+  sm: 20,
+  md: 24,
+  lg: 28,
+  xl: 32,
+};
+
+const checkedPadding = {
+  xs: 7.5,
+  sm: 10,
+  md: 11.5,
+  lg: 13,
+  xl: 15,
+};
+
+export default createStyles(
+	(theme, { color, radius, size, transitionDuration, variant }: ChipStyleParams, getRef) => {
+		return {
+      root: {},
+
+      label: {
+        ref: getRef('label'),
+        boxSizing: 'border-box',
+        color: theme.colors.black.value,
+        darkMode: {
+          color: theme.fn.themeColor('dark', 0)
+        },
+        display: 'inline-block',
+        alignItems: 'center',
+        userSelect: 'none',
+        border: '1px solid transparent',
+        borderRadius: theme.fn.radius(radius),
+        height: theme.fn.size({ size, sizes }),
+        fontSize: theme.fn.size({ size, sizes: theme.fontSizes }),
+        lineHeight: `${theme.fn.size({ size, sizes }) - 2}px`,
+        paddingLeft: theme.fn.size({ size, sizes: padding }),
+        paddingRight: theme.fn.size({ size, sizes: padding }),
+        cursor: 'pointer',
+        whiteSpace: 'nowrap',
+        transition: `border-color ${transitionDuration}ms ease, background-color ${transitionDuration}ms ease`,
+        WebkitTapHighlightColor: 'transparent',
+
+        '&.outline': {
+          darkMode: {
+            border: `1px solid ${theme.fn.themeColor('dark', 4)}`,
+            backgroundColor: theme.fn.themeColor('dark', 6),
+          },
```

Comment:
Bug caught! 🐛 There seems to be a bug with the current logic of dark mode ('sub-styles' do not sanitize the darkMode object - will investigate a solution later cc: @Brisklemonade), the darkMode styles inside `&.outline` and others will need to be changed to the previous solution of `[${theme.dark} &]: { ...`, sorry 😕

Reply:
Ah didn't realize that, the current `createStyles` has plans for depreciation and will be replaced with a newer better version.